src/software/backend/simulation/simulator_robot_test.cpp

```diff
@@ -3,39 +3,840 @@
+    // Check the ball has stuck to the dribbler
+    EXPECT_LT((simulator_ball->velocity() - Vector(0, 0)).length(), 0.001);
+    Point dribbling_point = getDribblingPoint(simulator_robot);
+    EXPECT_LT((simulator_ball->position() - dribbling_point).length(), 0.01);
+}
+
+TEST_F(SimulatorRobotTest, test_dribble_ball_while_moving_backwards)
+{
+    Robot robot(0, Point(0, 0), Vector(-0.5, 0), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Ball ball(Point(0.15, 0), Vector(-3, 0), Timestamp::fromSeconds(0));
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    EXPECT_LT((simulator_ball->velocity() - Vector(-3, 0)).length(), 0.001);
+
+    // We use an arbitrarily large number here for speed
+    simulator_robot->setDribblerSpeed(10000);
+
+    // Simulate for 2 seconds
+    for (unsigned int i = 0; i < 120; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    // Check the ball has stuck to the dribbler
+    EXPECT_LT((simulator_ball->velocity() - simulator_robot->velocity()).length(), 0.001);
+    Point dribbling_point = getDribblingPoint(simulator_robot);
+    EXPECT_LT((simulator_ball->position() - dribbling_point).length(), 0.01);
+}
+
+TEST_F(SimulatorRobotTest, test_dribble_ball_while_moving_forwards)
+{
+    Robot robot(0, Point(0, 0), Vector(0.5, 0), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Ball ball(Point(0.15, 0), Vector(0, 0), Timestamp::fromSeconds(0));
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    // We use an arbitrarily large number here for speed
+    simulator_robot->setDribblerSpeed(10000);
+
+    // Simulate for 1 second
+    for (unsigned int i = 0; i < 60; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    // Check the ball has stuck to the dribbler
+    EXPECT_LT((simulator_ball->velocity() - simulator_robot->velocity()).length(), 0.02);
+    Point dribbling_point = getDribblingPoint(simulator_robot);
+    EXPECT_LT((simulator_ball->position() - dribbling_point).length(), 0.01);
+}
+
+TEST_F(SimulatorRobotTest, test_dribble_ball_while_moving_spinning_in_place)
+{
+    Robot robot(0, Point(0, 0), Vector(0.0, 0), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Ball ball(Point(0.15, 0), Vector(-0.5, 0), Timestamp::fromSeconds(0));
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    // We use an arbitrarily large number here for speed
+    simulator_robot->setDribblerSpeed(10000);
+
+    // Simulate for 0.5 second so the ball makes contact with the dribbler
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    // Check the ball has stuck to the dribbler
+    EXPECT_LT((simulator_ball->velocity() - Vector(0.0, 0)).length(), 0.005);
```

Comment:
same as above

Reply:
Intentionally redundant for explicitness, so it matches the initial state of the robot specified at the start of the test